pkg/sbom/scanner/scanner.go

```diff
@@ -132,15 +132,28 @@ func (s *Scanner) start(ctx context.Context) {
 				_ = log.Errorf("invalid collector '%s'", request.Collector())
 			}
 
-			if err := s.enoughDiskSpace(collector.Options()); err != nil {
-				var imgMeta *workloadmeta.ContainerImageMetadata
-				if store := workloadmeta.GetGlobalStore(); store != nil {
-					img, err := store.GetImage(request.ID())
-					if err != nil {
-						imgMeta = img
-					}
+			var imgMeta *workloadmeta.ContainerImageMetadata
+			if collector.Type() == collectors.ContainerImageScanType {
+				store := workloadmeta.GetGlobalStore()
+				// The store should never be nil as workloadmeta is emitting the scan request
+				if store == nil {
+					log.Errorf("workloadmeta store is not initialized")
+					s.scanQueue.AddRateLimited(request)
+					s.scanQueue.Done(request)
+					continue
 				}
+				img, err := store.GetImage(request.ID())
+				if err != nil || img == nil {
+					// It can happen if the image is deleted between the time it was enqueued and the time it was processed
+					log.Debugf("failed to get image metadata for image id %s: %s", request.ID(), err)
```

Comment:
Nit: should we print this only when err != nil?
`img == nil` is expected sometimes as you mentioned in the comment.

Reply:
I think that if it should either return something or `errors.NewNotFound(id)`. I added the `img == nil` check for extra safety.

Reply:
Right. Then should we skip the log if the error is a not found one?
We already have a helper defined to check that in errors.go: `IsNotFound(err error)`

Reply:
I think that it could help to understand if there is an issue where for some reasons we can't find the image in workloadmeta when we should. It also can help to understand if we send a scan request and somehow we don't get a scan result and we don't know the outcome because there is a bug in workloadmeta.